pjmedia/src/pjmedia/stream.c

```diff
@@ -1893,8 +1893,8 @@ static void on_rx_rtp( pjmedia_tp_cb_param *param)
      */
     check_pt = (hdr->pt != stream->rx_event_pt) && PJMEDIA_STREAM_CHECK_RTP_PT;
     pjmedia_rtp_session_update2(&channel->rtp, hdr, &seq_st, check_pt);
-#if !PJMEDIA_STREAM_CHECK_RTP_PT 
-    if (!check_pt && hdr->pt != channel->rtp.out_pt &&
+#if PJMEDIA_STREAM_CHECK_RTP_PT 
+    if (check_pt && hdr->pt != channel->rtp.out_pt &&
```

Comment:
I changed the `check_pt` here to match how the check is made in rtp.c, I'm not sure what the original intention was though on this particular line.

Reply:
To me, the intention is to assign `flag.badpt` when `PJMEDIA_STREAM_CHECK_RTP_PT` is not set. Note that if `PJMEDIA_STREAM_CHECK_RTP_PT` is set, `pjmedia_rtp_session_update2()` will handle the `badpt` assignment. So I think the existing code is already correct.

Reply:
The problem is that merely setting

```c
#define PJMEDIA_STREAM_CHECK_RTP_PT 0
```

in the config_site.h alone is not enough because with that defined that way, the few lines above effectively become:


```c
check_pt = PJ_FALSE;
// does not mark badpt
pjmedia_rtp_session_update2(&channel->rtp, hdr, &seq_st, check_pt);
// erroneously marks badpt, despite `check_pt` being false
if (hdr->pt != channel->rtp.out_pt &&
	hdr->pt != stream->rx_event_pt) 
```

Reply:
Yes, then the `badpt` will be set to 1, that's actually the purpose of the `if !PJMEDIA_STREAM_CHECK_RTP_PT` block. Setting the `badpt` without `bad` won't cause the RTP packet to be dropped, it is just for logging purpose.